protocols/smtp/src/main/java/org/apache/james/protocols/smtp/SMTPSessionImpl.java

```diff
@@ -41,8 +42,12 @@ public class SMTPSessionImpl extends ProtocolSessionImpl implements SMTPSession
 
     public SMTPSessionImpl(ProtocolTransport transport, SMTPConfiguration config) {
         super(transport, config);
-        relayingAllowed = Optional.ofNullable(getRemoteAddress())
-            .flatMap(addres -> Optional.ofNullable(addres.getAddress()))
+        relayingAllowed = computeRelayingAllowed(config);
+    }
+
+    private Boolean computeRelayingAllowed(SMTPConfiguration config) {
+        return Optional.ofNullable(getRemoteAddress())
+            .flatMap(address -> Optional.ofNullable(address.getAddress()))

```

Comment:
```suggestion
            .map(address -> address.getAddress())
```

as `Optional.map` already handles null case:

![image](https://github.com/apache/james-project/assets/55171818/c391eb43-f450-4b03-a68c-9bed1e7a5339)

(I did not know that before)

Reply:
I prefer being explicit to be honnest

Reply:
But thanks for sharing.